### DISCUSSION TITLE
Issues running PnP PowerShell Scripts using Task Scheduler

### DISCUSSION THREAD

**Discussion by ghost** [pnp/powershell#228] [Q&A]
Hello,

I've powershell script scheduled using task scheduler to execute on SharePoint Online lists. This script goal is to clear the items in the list (A), read through the lists on sub-sites (S1, S2. S3, ....) and display the items from these sub-sites lists to list (A). 

The script is executing as expected on running manually, but when scheduled to run via Task Scheduler sometimes (5%) it works as expected and sometimes (95%) it completes the execution without feeding any data to list (A). I worked with Microsoft and they routed me here for help!

- My first question is about Error 429. How do I overcome this? What command should I use and where to input this in my code?
- Second question: Most of the items the scripts on scheduler fails to run the entire code and doesn't show any error on the logs.    How to mitigate this issue?

I would appreciate your help and I can provide you with more details if needed.

**Comment by erwinvanhunen** [comment]
Which version of PnP Powershell are you using? 429 errors means you're being throttled by the server. The new PnP PowerShell (`Install-Module -Name PnP.PowerShell`, see for more info here: https://pnp.github.io/powershell) follows the guidelines from MS on on how to handle throttling, e.g. you should not notice it occurring.

**Reply by ghost** [reply]
Using Windows Powershell ISE build: 18362, Revision: 1171
